DOC: discuss legacy backends (#146)
audeering/audbackend: docs/legacy.rst

.. set temporal working directory
.. jupyter-execute::
    :hide-code:

    import os
    import audeer

    _cwd_root = os.getcwd()
    _tmp_root = audeer.mkdir(os.path.join('docs', 'tmp'))
    os.chdir(_tmp_root)

.. _legacy-backends:

Legacy backends
===============

The file structure on the backend
has changed for
:class:`audbackend.FileSystem`
and :class:`audbackend.Artifactory`
in version 1.0.0
of :mod:`audbackend`.

> Before a file ``/sub/file.txt``

Reviewer: ```suggestion
Before,
a file ``/sub/file.txt``
```

with version ``1.0.0``
was stored under

.. code-block::

    /sub/file/1.0.0/file-1.0.0.txt

> and is now stored under

Reviewer: ```suggestion
Now it is stored under
```

.. code-block::

    /sub/1.0.0/file.txt

> If you need to continue
> with the old file structure
> in your application,
> you can configure :mod:`audbackend` accordingly
> by calling the hidden method
> ``_use_legacy_file_structure()``
> after instantiating the backend.

Reviewer: ```suggestion
To force the old file structure
call the hidden method
``_use_legacy_file_structure()``
after instantiating the backend.
We recommend this 
for existing repositories
that store files
under the old structure.
```
